server/src/com/thoughtworks/go/config/GoFileConfigDataSource.java

```diff
@@ -221,9 +221,13 @@ private void writeToConfigXmlFile(String content) {
             try {
                 LOGGER.info(String.format("[Configuration Changed] Saving updated configuration."));
                 String configAsXml = configAsXml(modifiedConfig, true);
+                String md5 = CachedDigestUtils.md5Hex(configAsXml);
+                MagicalGoConfigXmlLoader.setMd5(modifiedConfig, md5);
```

Comment:
I see you'd said: "Also, MD5 wasn't being set during config save api calls". Ok, fine.